### PR TITLE
Http Event Listener retry for codes not in [200, 300) and expections, rertry delay calculation fix, better logging

### PR DESCRIPTION
**Pull request by mosiac1**
I have been running into problems while using the event listener. Problems are usually one of:

1. Broken Pipe Exceptions, which are irregular and usually occur less than 2 times per hour. These aren't critical but cause dropped events. (Cause of these I suspect might be bad timing between trino and the receiving server in regards to timeouts) here is an [exception](https://gist.github.com/mosiac1/657c2a2555791637a24d3089e7aad7f5);

2. Regular timeouts. There are cases where most of the plugin's attempts to send events will end with a timeout (from the http-client, not from the receiving server). The cause of these issues I haven't been able to track down yet. What I know is that it's probably not the fault of the ingest server because when the plugin keeps timing-out other requests to that server went through correctly (event from the same machine as trino). This usually happens when there are high loads on trino.

The changes this PR implements are simple and self-explanatory from the title. 

These should fix problem 1. and help with tracking down problem 2. 

Any ideas regarding problem 2 are greatly appreciated!

### CODE REVIEW

File: plugin/trino-http-event-listener/src/main/java/io/trino/plugin/httpquery/HttpEventListener.java [hunk 1]
```diff
@@ -82,27 +82,27 @@ public HttpEventListener(HttpEventListenerConfig config, @ForHttpEventListener H
     public void queryCreated(QueryCreatedEvent queryCreatedEvent)
     {
         if (config.getLogCreated()) {
-            sendLog(queryCreatedEvent);
+            sendLog(queryCreatedEvent, queryCreatedEvent.getMetadata().getQueryId());
```
Comment (losipiuk) [inline plugin/trino-http-event-listener/src/main/java/io/trino/plugin/httpquery/HttpEventListener.java:85]: Commit messge `Add more logging the event listener` -> `Add more logging in the HTTP event listener`

File: plugin/trino-http-event-listener/src/main/java/io/trino/plugin/httpquery/HttpEventListener.java [hunk 2]
```diff
@@ -125,19 +125,33 @@ public void onSuccess(StatusResponse result)
...
 
                                 if (!(result.getStatusCode() >= 200 && result.getStatusCode() < 300)) {
-                                    log.error("Received status code %d from ingest server URI %s; expecting status 200", result.getStatusCode(), request.getUri());
+                                    log.warn("QueryId = \"%s\", attempt = %d/%d, URL = %s | Received status code %d from ingest server; expecting status 200",
+                                            queryId, attempt + 1, config.getRetryCount(), request.getUri().toString(),
+                                            result.getStatusCode(), request.getUri().toString());
+                                    return;
                                 }
+
+                                log.info("QueryId = \"%s\", attempt = %d/%d, URL = %s | Query event delivered successfully",
```
Comment (losipiuk) [inline plugin/trino-http-event-listener/src/main/java/io/trino/plugin/httpquery/HttpEventListener.java:139]: feels to verbose. I think it should be `debug`

**Review by losipiuk — COMMENTED** [review]
LGTM - minor comments

**Comment by mosiac1** [comment]
Thanks for the review! I implemented the requested changes.

I also found a bug in the next delay calculation and fixed that as well (+ changed PR title to include that).

File: plugin/trino-http-event-listener/src/main/java/io/trino/plugin/httpquery/HttpEventListener.java (hunk 2 above)
Comment (losipiuk) [reply plugin/trino-http-event-listener/src/main/java/io/trino/plugin/httpquery/HttpEventListener.java]: I see you changed that in latter commit but it should be `debug` already in "Add more logging to the HTTP event listener"

**Review by losipiuk — APPROVED** [review]

File: plugin/trino-http-event-listener/src/main/java/io/trino/plugin/httpquery/HttpEventListener.java (hunk 2 above)
Comment (losipiuk) [reply plugin/trino-http-event-listener/src/main/java/io/trino/plugin/httpquery/HttpEventListener.java]: Please do. We want clean commit history when possible. I will merge when CI passes.

File: plugin/trino-http-event-listener/src/main/java/io/trino/plugin/httpquery/HttpEventListener.java [hunk 3]
```diff
@@ -124,22 +124,15 @@ public void onSuccess(StatusResponse result)
                             {
                                 verify(result != null);
 
-                                if (result.getStatusCode() >= 500 && attempt < config.getRetryCount()) {
+                                if (!(result.getStatusCode() >= 200 && result.getStatusCode() < 300) && attempt < config.getRetryCount()) {
```
Comment (hashhar) [inline plugin/trino-http-event-listener/src/main/java/io/trino/plugin/httpquery/HttpEventListener.java:127]: I'm not sure how would retrying any 4xx error ever succeed? Wouldn't we end up retrying until retry attempts are exhausted?

Comment (mosiac1) [reply plugin/trino-http-event-listener/src/main/java/io/trino/plugin/httpquery/HttpEventListener.java:127]: Yes, we would retry until attempts are exhausted.

Comment (mosiac1) [reply plugin/trino-http-event-listener/src/main/java/io/trino/plugin/httpquery/HttpEventListener.java:127]: I think it has its use-cases (which admittedly are edge-cases) and it's better to be safe than drop data. This doesn't run on the query execution threads so no direct slowdown.

**Pull request merged**